src/gui/editors/SongEditor.cpp

```diff
@@ -1125,6 +1228,51 @@ void SongEditorWindow::lostFocus()
 }
 
 
+void SongEditorWindow::showZoomingSliderFloat()
+{
+	m_zvsStatus->moveGlobal(m_zoomingSlider, QPoint(m_zoomingSlider->width() + 2, -2));
+	m_zvsStatus->show();
+	updateZoomingSliderFloat(m_editor->scaleFunction(m_editor->m_zoomingLogModel->value(), false));
+}
+
+void SongEditorWindow::updateZoomingSliderFloat(int newVal)
+{
+	// calculate the min zoom based on song length and song editor size
+	int minZoom = m_editor->calculateMinZoom();
+
+	newVal = m_editor->scaleFunction(newVal, true);
+	newVal = std::max(newVal, minZoom);
+	if (newVal <= minZoom) { m_zoomingSlider->setValue(m_editor->scaleFunction(minZoom, false)); }
+
+	m_editor->m_zoomingLogModel->setValue(newVal);
+	m_zvsStatus->setText(tr("Value: %1%").arg(newVal));
+}
+
+void SongEditorWindow::hideZoomingSliderFloat()
+{
+	m_zvsStatus->hide();
+}
+
+// key events are set at SongEditorWindow level to make them work in all the window and not only on SongEditor area.
+void SongEditorWindow::keyPressEvent(QKeyEvent* ke)
+{
+	if (ke->key() == Qt::Key_0 && ke->modifiers() & Qt::ControlModifier)
+	{
+		m_editor->m_zoomingLogModel->setValue(m_editor->m_zoomingLogModel->initValue());
+		m_editor->m_zoomingLinearModel->setValue(m_editor->scaleFunction(m_editor->m_zoomingLogModel->initValue(), false));
+	}
+	else
+	{
+		QWidget::keyPressEvent(ke);
+	}
+}
+
+void SongEditorWindow::keyReleaseEvent(QKeyEvent* ke)
+{
+	if (m_editor->m_zvsStatus->isVisible()) { m_editor->m_zvsStatus->hide(); }
```

Comment:
You could clear `m_iniBar` here, but I still think it's a better idea to just skip that

Reply:
discussed in a previous commentary